04_verify.sh

```diff
@@ -0,0 +1,184 @@
+    echo "$((FAILS+1))"
+  fi
+}
+
+# Compare the two inputs and log
+differs(){
+  if [[ "${1}" != "${2}" ]]; then
+    echo "OK - ${3}" >&3
+    echo "${FAILS}"
+  else
+    echo "FAIL - ${3}" >&3
+    echo "$((FAILS+1))"
+  fi
+}
+
+function check_bm_hosts() {
+    while read -r name address user password mac; do
+      BM_VMNAME="${name//-/_}"
+      # Verify BM host exists
+      echo "$BM_HOSTS" | grep -w "${name}"  > /dev/null
+      FAILS="$(process_status $? "${name} Baremetalhost exist")"
+
+      BM_HOST="$(echo "${BM_HOSTS}" | jq ' .items[] | select(.metadata.name=="'"${name}"'" )')"
+
+      # Verify addresses of the host
+      FAILS="$(equals "$(echo "${BM_HOST}" | jq -r '.spec.bmc.address')" \
+        "${address}" "${name} Baremetalhost address correct")"
+      FAILS="$(equals "$(echo "${BM_HOST}" | jq -r '.spec.bootMACAddress')" \
+        "${mac}" "${name} Baremetalhost mac address correct")"
+
+      # Verify BM host status
+      FAILS="$(equals "$(echo "${BM_HOST}" | jq -r '.status.operationalStatus')" \
+        "OK" "${name} Baremetalhost status OK")"
+
+      # Verify credentials exist
+      CRED_NAME="$(echo "${BM_HOST}" | jq -r '.spec.bmc.credentialsName')"
+      CRED_SECRET="$(kubectl get secret "${CRED_NAME}" -n metal3 -o json | jq '.data')"
+      FAILS="$(process_status $? \
+        "${name} Baremetalhost credentials secret exist")"
+
+      # Verify credentials correct
+      FAILS="$(equals "$(echo "${CRED_SECRET}" | jq -r '.password' | base64 --decode)" \
+        "${password}" "${name} Baremetalhost password correct")"
+      FAILS="$(equals "$(echo "${CRED_SECRET}" | jq -r '.username' | base64 --decode)" \
+        "${user}" "${name} Baremetalhost user correct")"
+
+      # Verify the VM was created
+      echo "$BM_VMS "| grep -w "${BM_VMNAME}"  > /dev/null
+      FAILS="$(process_status $? "${name} Baremetalhost VM exist")"
+
+      BM_VM_IFACES="$(virsh domiflist "${BM_VMNAME}")"
+      for bridge in ${BRIDGES}; do
+        echo "$BM_VM_IFACES" | grep -w "${bridge}"  > /dev/null
+        FAILS="$(process_status $? \
+          "${name} Baremetalhost VM interface ${bridge} exist")"
+      done
+      echo "" >&3
+    done
+    echo "${FAILS}"
+}
+
+#Verify that a resource exists in a type
+function check_k8s_entity() {
+  for name in ${2}; do
+    # Check entity exists
+    ENTITY="$(kubectl --kubeconfig "${KUBECONFIG}" get "${1}" "${name}" -n metal3 -o json)"
+    FAILS="$(process_status $? "${1} ${name} created")"
+    # Check the replicas
+    FAILS="$(equals "$(echo "${ENTITY}" | jq -r '.status.readyReplicas')" \
+      "$(echo "${ENTITY}" | jq -r '.status.replicas')" \
+      "${name} ${1} replicas correct")"
+  done
+  echo "" >&3
+  echo "${FAILS}"
+}
+
+#Verify that a resource exists in a type
+function check_k8s_rs() {
+  for name in ${1}; do
+    # Check entity exists
+    ENTITY="$(kubectl --kubeconfig "${KUBECONFIG}" get replicasets -l name="${name}" -n metal3 -o json | jq '.items[0]')"
```

Comment:
you can put this line
kubectl --kubeconfig "${KUBECONFIG}"
to KUBECTL
to reduce clutter.